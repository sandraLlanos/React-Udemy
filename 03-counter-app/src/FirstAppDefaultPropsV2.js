import React from 'react';
import PropTypes from 'prop-types';

const FirstAppDefaultPropsV2 = ({ title, subtitle }) => {

    return (
        <>
            <h1> {title} </h1>
            <p> {subtitle} </p>
        </>
    )

}

FirstAppDefaultPropsV2.propTypes = {
    // greeting: PropTypes.string,
    // greeting: PropTypes.string.isRequired,
}
FirstAppDefaultPropsV2.defaultProps = {
    title: "I'm a title",
    subtitle: "I'm a subtitle"
}

export default FirstAppDefaultPropsV2;