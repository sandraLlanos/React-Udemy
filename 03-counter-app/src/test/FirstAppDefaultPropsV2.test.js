import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import FirstAppDefaultPropsV2 from "../FirstAppDefaultPropsV2";

describe('test in <FirstAppDefaultPropsV2 /> component', () => {
    
    xtest('should show the message "Hello, I am Goku"', () => {
        
        const saludo = "Hello";
        const { getByText } = render(<FirstAppDefaultPropsV2 greeting={saludo} />);
        expect(getByText(saludo)).toBe();

    })

    test('should show <FirstAppDefaultPropsV2> correctly', () => {
        const saludo = "Hello";
        const wrapper = shallow(<FirstAppDefaultPropsV2 greeting={saludo} />)
        expect( wrapper ).toMatchSnapshot();

    })
    
    

})
