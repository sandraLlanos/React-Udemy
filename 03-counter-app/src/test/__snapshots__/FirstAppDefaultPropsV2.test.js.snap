// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <FirstAppDefaultPropsV2 /> component should show <FirstAppDefaultPropsV2> correctly 1`] = `
<Fragment>
  <h1>
     
    I'm a title
     
  </h1>
  <p>
     
    I'm a subtitle
     
  </p>
</Fragment>
`;
